### PR TITLE
feat(thread_id): add logging adapter for thread ids

### PR DESCRIPTION
**Pull request by anjakefala**
Making this available for testing! This is a prototype, that works in isolation, but I would like to ensure it has a good relationship with the other logging.

### CODE REVIEW

**Comment by jrs65** [comment]
Looks awesome. Doesn't build but I think that can be fixed up while Rick is merging.

**Comment by nritsche** [comment]
@anjakefala Can you add the comment Richard requested?

File: comet/broker.py [hunk 1]
```diff
@@ -31,6 +33,67 @@
...
+# reqfilter = RequestFilter(request_id)
+# formatter = RequestFormatter(request_id)
+# formatter = logging.Formatter('%(asctime)s - %(name)s - %(levelname)s - %(message)s')
+formatter = RequestFormatter(
+    request_id,
+    "%(asctime)s [%(process)d] [%(name)s:%(levelname)s] [request=%(request_id)s] %(message)s",
+)
+syslog.setFormatter(formatter)
+logging.getLogger("comet").setLevel(logging.DEBUG)
+# logging.getLogger("").addFilter(reqfilter)
```
Comment (nritsche) [inline comet/broker.py]: remove old comment

File: comet/broker.py [hunk 2]
```diff
@@ -31,6 +33,67 @@
...
+logging.getLogger("").addHandler(syslog)
```
Comment (nritsche) [inline comet/broker.py]: When I merged @jrs65's branches into this, I saw you had passed `"comet"` to `getLogger` and Richard removed it. I don't know what we want in the end. Do you?

Comment (anjakefala) [reply comet/broker.py]: I do!

Richard wants this to get associated with the root logger. Which means that the correct thing to do is to pass `''` to `getLogger`.

Comment (anjakefala) [reply comet/broker.py]: (I agree with associating this with the root logger. =))

**Comment by anjakefala** [comment]
@nritsche @jrs65 I added the requested comments, and erased the commented-out code!

There seems to be a lot more stuff in this PR than there was before... =) I leave it to your discretion for whether that is expected and okay to merge.

**Comment by nritsche** [comment]
> @nritsche @jrs65 I added the requested comments, and erased the commented-out code!
Thanks!

> There seems to be a lot more stuff in this PR than there was before... =) I leave it to your discretion for whether that is expected and okay to merge.
Yeah, it's just that Richards branches where based on this, so I thought the easiest way is to just merge it back together this way.

File: .travis.yml [hunk 3]
```diff
@@ -59,7 +59,8 @@ script:
     - pytest -s
 
     # Clone kotekan and run some of its tests
-    - ./kotetest.sh
+    - mkdir skyfield_data
+    - CAPUT_SKYFIELD_PATH=skyfield_data ./kotetest.sh
```
Comment (anjakefala) [inline .travis.yml]: noooooo

Comment (nritsche) [reply .travis.yml]: I actually don't think this was the problem. One of the two CI builds still failed. It doesn't say why and I can't reproduce...

Comment (anjakefala) [reply .travis.yml]: test_dataset_broker.py:59: AssertionError

============================== 1 failed in 3.26s ===============================

The command "CAPUT_SKYFIELD_PATH=skyfield_data ./kotetest.sh" exited with 1.

Comment (anjakefala) [reply .travis.yml]: Oh, yeah, this is a super vague assert

```
>               assert call([producer_path, "--", port]) == 0

E               AssertionError: assert -4 == 0

E                +  where -4 = call(['../build/tests/boost/dataset_broker_producer', '--', '46083'])
```

Comment (nritsche) [reply .travis.yml]: Need to print stdout AND stderr also in case it fails...

**Comment by anjakefala** [comment]
:crossed_fingers:

**Comment by anjakefala** [comment]
```
[2019-12-12 20:12:04 +0000] - (sanic.access)[INFO][127.0.0.1:56290]: POST http://localhost:4444/send-state  200 20

2019-12-12 20:12:04,069 [8332] [sanic.access:INFO] [request=784736408626] 

No output has been received in the last 10m0s, this potentially indicates a stalled build or something wrong with the build itself.

Check the details on how to adjust your build configuration on: https://docs.travis-ci.com/user/common-build-problems/#build-times-out-because-no-output-was-received

The build has been terminated

```

**Comment by nritsche** [comment]
```
Illegal instruction (core dumped)
```
Progress?

**Comment by anjakefala** [comment]
What did you do?? @nritsche

**Comment by nritsche** [comment]
I just executed that test binary directly instead of letting python do it (which hid the output).

**Comment by anjakefala** [comment]
It seems...happy.

Do we merge? =)

**Comment by nritsche** [comment]
Oh I didn't noticed it passed. I think either travis is giving us different nodes some times or they are fiddling around with their settings atm.

**Comment by nritsche** [comment]
Let me fix up the changes I was making on kotekan side. If it keeps passing, yes!

**Pull request merged**